pkg/agent/util/net_linux.go

```
@@ -376,3 +378,103 @@ func renameHostInterface(oriName string, newName string) error {
+		return true, nil
+	}); err != nil {
+		return "", false, fmt.Errorf("failed waiting for internal port to show up: %v", err)
+	}
+
+	localLink, err := netlink.LinkByName(linkConfig.Name)
+	if err != nil {
+		return "", false, err
+	}
+	if _, _, err = SetLinkUp(linkConfig.Name); err != nil {
+		return "", false, fmt.Errorf("failed to set link up: %v", err)
+	}
+
+	// Check if interface is configured with an IPv6 address: if it is, we need to ensure that IPv6
+	// is enabled on the OVS internal port as we need to move all IP addresses over.
+	for _, ip := range linkConfig.IPs {
+		if ip.IP.To4() == nil {
+			klog.InfoS("Interface has IPv6 address, ensuring that IPv6 is enabled on bridge local port", "port", linkConfig.Name)
+			if err := EnsureIPv6EnabledOnInterface(linkConfig.Name); err != nil {
+				klog.ErrorS(err, "Failed to ensure that IPv6 is enabled on bridge local port, moving uplink IPs to bridge is likely to fail", "port", linkConfig.Name)
+			}
+			break
+		}
+	}
+
+	if err = ConfigureLinkAddresses(localLink.Attrs().Index, linkConfig.IPs); err != nil {
+		return "", false, err
+	}
+	if err = ConfigureLinkAddresses(linkConfig.Index, nil); err != nil {
+		return "", false, err
+	}
+	// Restore the host routes which are lost when moving the network configuration of the
+	// uplink interface to OVS bridge interface.
+	if err = ConfigureLinkRoutes(localLink, linkConfig.Routes); err != nil {
+		return "", false, err
+	}
+
+	return bridgedName, false, nil
+}
+
+func RestoreHostInterfaceConfiguration(brName string, linkConfig *config.AdapterNetConfig) {
+	klog.InfoS("Restoring bridge config to uplink...")
+	if linkConfig.Name != "" {
+		bridgedName := GenerateUplinkInterfaceName(linkConfig.Name)
+
+		if err := DeleteOVSPort(brName, linkConfig.Name); err != nil {
+			klog.ErrorS(err, "Delete OVS port failed", "port", linkConfig.Name)
+		}
+		if err := DeleteOVSPort(brName, bridgedName); err != nil {
```

To Wenying's point, if the uplink port is not created, which means the physical interface has not been moved to the bridge, we should not restore the IPs & routes from the OVS internal port to the interface?

---

And for all these operations, we should just ignore the not-exist error, as the func can be called after a partial failure happened at interface moving, and so possible the interface is not moved and the OVS port is not created.

---

Added a check to exit early, if the bridged port doesn't exists (eth0~).

> And for all these operations, we should just ignore the not-exist error
DeleteOVSPort runs the del command with `--if-exists` flag, so those errors will be ignored. And also the first condition would now prevent this.